pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -177,3 +210,88 @@ func unmarshalProviderSpec(spec *runtime.RawExtension) (*v1alpha1.AzureMachinePr
 	klog.V(6).Infof("Found ProviderSpec: %+v", config)
 	return &config, nil
 }
+
+func updateCluster(credentialsSecret *apicorev1.SecretReference, scope *Scope) error {
+	if credentialsSecret == nil {
+		return errors.New("provided empty credentials secret")
+	}
+
+	secretName := fmt.Sprintf("%s/%s", credentialsSecret.Namespace, credentialsSecret.Name)
+
+	cfg := controllerconfig.GetConfigOrDie()
```

Comment:
`GetConfigOrDie()` will call `os.Exit(1)` if there's an error. That seems kind of harsh here. Any reason not to use `GetConfig()` and return the error? Same goes for the other few spots in this file that create a new client.